packages/graph/hash_graph/lib/graph/src/types/uri.rs

```diff
@@ -0,0 +1,81 @@
+use std::{fmt, result::Result as StdResult, str::FromStr};
+
+use error_stack::{Context, IntoReport, Report, Result, ResultExt};
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+pub type BaseId = String;
+pub type BaseIdRef = str;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Uri {
```

Comment:
In a similar vein to above, I think we should call this `VersionedUri` to be more consistent with the BP Spec (it's also less ambiguous)